recipes/salmon/salmon_cmake.patch

```diff
@@ -0,0 +1,20 @@
+--- CMakeLists.txt
++++ CMakeLists.txt
+@@ -339,7 +339,7 @@ if (BOOST_RECONFIGURE)
+     set(CMAKE_PREFIX_PATH ${CMAKE_CURRENT_SOURCE_DIR}/external/install)
+     set(Boost_INCLUDE_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/external/install/include)
+     set(Boost_LIBRARY_DIRS ${CMAKE_CURRENT_SOURCE_DIR}/external/install/lib)
+-    find_package(Boost 1.59.0 COMPONENTS iostreams filesystem system timer chrono program_options REQUIRED)
```

Comment:
We are also using boost 1.67, could this be a problem as well? https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/master/recipe/conda_build_config.yaml#L243

Reply:
Hi @bgruening,

  Newer should not be a problem, 1.59 is a lower bound.  If the user doesn't have boost available, we fetch 1.66.  I *believe* that CMake implements a >= policy for these versions, so the user must have version >= 1.59.0.  The issue is clearly related to finding these icu58-related symbols during linking.  I recall having an issue like this before, where the solution was to pin a specific version of icu & boost --- but now conda is using the newer version of icu (58) that we'd pinned by default, so I think the issue must be different.

Reply:
No current boost is not compiled with a recent GCC. conda-forge is still rebuilding :(
Sorry.

Reply:
ahhhh ok --- so I think this might make more sense now.  I wonder if this is the cause of a number of other failures, as any package that is building on cb3 (and hence, the newer gcc), but not _explicitly_ turning on the old ABI, should have issues.

Reply:
The newer GCC just comes in for salmon by adding this file to the salmon dir: https://github.com/bioconda/bioconda-recipes/blob/salmon/recipes/salmon/conda_build_config.yaml
So all (there are a handful other exceptions) other recipes do still build against old GCC until conda-forge has rebuild first.

Reply:
Ahh, so, salmon only need c++11 right now.  Could we revert to gcc 4.8.5 or whatever for this release? Would that fix things?

Reply:
Id be happy to give it a try but removing the file seems to anger the linter.

Reply:
@bgruening , 

  Any idea why I might be seeing this from the linter?

```
0	recipes/salmon	should_use_compilers	"{'should_use_compilers': True, 'fix': 'use {{ compiler(""c"") }} or other new-style compilers'}"	"use {{ compiler(""c"") }} or other new-style compilers"	True
```